setup.sh

```diff
@@ -87,7 +87,7 @@ function check_dependencies()
 
   while IFS='' read -r package; do
     pip list | grep -F "$package" > /dev/null
-    [[ "$?" != 0 ]] && pip_package_list="$package $pip_package_list"
+    [[ "$?" != 0 ]] && pip_package_list="\"$package\" $pip_package_list"
```

Comment:
Why do we need `\"`? Is it because of `jinja2>=3`?

Reply:
Thats right. `pip install jinja2>=3` like command will make the shell exec `pip install jinja2` and put its input into a `=3` file.

Reply:
Consider using `pip freeze -r <dependencies file>` instead of the whole `while` loop :)

```
# Get the list of uninstalled dependencies.
# `pip freeze -r <dependencies file>` compares the installed modules to the dependencies file and outputs to stderr lines like:
# "WARNING: Requirement file [pip.dependencies] contains sphinx-book-theme, but package 'sphinx-book-theme' is not installed"
# `sed` gets the module name from those lines and `paste` joins them in a single line.
pip_package_list=$(pip freeze -r $DOCUMENTATION/dependencies/pip.dependencies 2>&1 1> /dev/null | sed -r 's/^.*contains (.*),.*not installed.*$/\1/' | paste --ser
ial -d' ')
```

Reply:
Thanks for your advice. But for this case, we need to upgrade jinja2, because the lower version of jinja2 can't get along well with sphinx-book-theme, resulting in CI errors. The current solution checks whether the package is installed and whether its version meets the requirements. :>